### ISSUE TITLE
Limit the local port range the onion service can bind to

### ISSUE THREAD

**Issue by nyxnor** [bisq-network/bisq2#2856]
Thanks to @alvasw and the Bisq2 team, it v2.1.0 is working on Whonix 17.

Limiting port range the onion service can bind to locally. This is specially useful when using Whonix.

Currently, it recommends to open [all ports in the firewall of the Workstation](https://www.whonix.org/wiki/Bisq#Firewall_Settings) to  be externally reachable via `EXTERNAL_OPEN_ALL=true`. That is not great.

It would be better to use the same method [OnionShare uses](https://www.whonix.org/wiki/OnionShare#Firewall_Settings), port ranges and tries different ones till it finds an unused port, which would be a better firewall with `EXTERNAL_OPEN_PORTS+=" $(seq 17600 17659) "`, with appropriate changes to the onion-grater profile ([onion share example](https://github.com/Whonix/onion-grater/blob/84c85c85f90eb36757c503864e1e979e7034dc43/usr/share/doc/onion-grater-merger/examples/40_onionshare.yml#L46)).

In other words, limit the `localPort` of:

https://github.com/bisq-network/bisq2/blob/283bcffd54f2a1d1fd9d78f43479552ab6af0870/network/tor/tor/src/main/java/bisq/tor/controller/TorControlProtocol.java#L87

to a range of possible ports and document it, so Whonix users can stay with a tight firewa...

**Comment by mvp1983** [comment]
I can take if @alvasw doesn't have time for it

**Comment by mvp1983** [comment]
Before I start, some questions:

 17600..17659 - it is a default range only for Whonix or for others OS too?
do we need some UX interface, to be able to set port range?

**Comment by HenrikJannsen** [comment]
I think its better @alvasw works on that.
